src/Datasource/Paging/NumericPaginator.php

```diff
@@ -274,7 +274,7 @@ protected function getQuery(RepositoryInterface $object, ?QueryInterface $query,
             }
 
             $query = $object->find($type, ...$args);
-        } elseif (!empty($options['finder'])) {
+        } elseif (!empty($options['finder']) && $options['finder'] !== 'all') {
```

Comment:
```suggestion
        } elseif ($options['finder'] !== 'all') {
```

Since we have a default value for `finder` the non-empty check is not necessary